atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbErrorDecoder.java

```diff
@@ -57,8 +64,14 @@ private Date parseRetryAfter(Response response) {
     }
 
     private boolean response503ButExceptionIsNotRetryable(Response response, Exception exception) {
-        boolean responseIs503 = response.status() == 503;
-        boolean isRetryableException = exception instanceof RetryableException;
-        return responseIs503 && !isRetryableException;
+        return (response.status() == 503) && !isExceptionIsRetryable(exception);
+    }
+
+    private boolean response429ButExceptionIsNotRetryable(Response response, Exception exception) {
+        return (response.status() == 429) && !isExceptionIsRetryable(exception);
+    }
```

Comment:
Would it make sense to parameterise the error code we're hoping to match? Don't feel strongly about this, in any case.